### ISSUE TITLE
Missing org.antlr.runtime package

### ISSUE THREAD

**Issue by gnl42** [eclipse-mylyn/org.eclipse.mylyn#65]
I see this when I launch Mylyn:

`!ENTRY org.eclipse.mylyn.reviews.tasks.dsl 4 0 2023-03-16 22:53:12.415
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: org.eclipse.mylyn.reviews.tasks.dsl [448]
  Unresolved requirement: Import-Package: org.antlr.runtime; version="[3.0.0,3.1.0)"
`

**Comment by gnl42** [comment]
It's not a build issue. It shows up when you launch Mylyn to debug/test 
the code

**Comment by BeckerFrank** [comment]
Did you use the launch config stored here (/org.eclipse.mylyn.development/Work)?

**Comment by BeckerFrank** [comment]
I think that the tbr (task based reviews) are not active and we can delete the whole folder.
Can not reproduce the launch error.

**Comment by gnl42** [comment]
Yes, Mylyn-Runtime

**Comment by BeckerFrank** [comment]
OK,
now I know why I can not reproduce this.

I use the Mylyn Setup from the Eclipse-Installer and here we do not import all the projects.
We exclude the following path
 - mylyn.builds/org.eclipse.mylyn.hudson.tests/testdata
- mylyn.reviews/tbr
- mylyn.docs
- /org.eclipse.mylyn.java.tests/target

so you need to close or remove the projects located in the excluded path.

Maybe you can use the Eclipse-Installer Mylyn setup the next time you install a new IDE.

**Comment by gnl42** [comment]
Hi Frank,
    I used the original Eclipse-Install Mylyn which pointed to the out 
of date repository so I imported the branch manually.
I'll re-install and see what happens

**Comment by BeckerFrank** [comment]
Hert the URL of the actual mylyn.setup https://raw.githubusercontent.com/eclipse-mylyn/org.eclipse.mylyn/main/org.eclipse.mylyn.releng/oomph/Mylyn.setup

**Comment by gnl42** [comment]
Have it working now

**Comment by gnl42** [comment]
So how do I make my development eclipse match what the .setup file does since I'm working with my forked repository?

**Comment by wimjongman** [comment]
Just setup the dev environment and then in the Git perspective, add your remote to the repo.

**Comment by ruspl-afed** [comment]
Is it still relevant @gnl42 ?

**Comment by gnl42** [comment]
HI @ruspl-afed, not really. 

I guess the question is why is the 
`We exclude the following path

    mylyn.builds/org.eclipse.mylyn.hudson.tests/testdata
    mylyn.reviews/tbr
    mylyn.docs
    /org.eclipse.mylyn.java.tests/target
`
part of the repository if they are not part of the build?

**Comment by BeckerFrank** [comment]
> HI @ruspl-afed, not really.
> 
> I guess the question is why is the `We exclude the following path
> 
> ```
> mylyn.builds/org.eclipse.mylyn.hudson.tests/testdata
> mylyn.reviews/tbr
> mylyn.docs
> /org.eclipse.mylyn.java.tests/target
> ```
> 
> ` part of the repository if they are not part of the build?

The exclude is only for the "Projects Import" so .project files in that part of the repository are ignored. 

- mylyn.builds/org.eclipse.mylyn.hudson.tests/testdata
  If you import  you get 65 errors in the Error View
- mylyn.reviews/tbr
  Task-based checks (tbr) were not really active, if I remember correctly (similar to the Bugzilla REST Connector)
- mylyn.docs
  we can remove this because this is now an other repository 
- /org.eclipse.mylyn.java.tests/target
  only relevant if you did a maven build an the import the projects

**Comment by gnl42** [comment]
Let me rephrase that.

Why are these "projects" around if it requires special knowledge to 
import the repository in order to have a clean
workspace?

What's the purpose of the projects?

George

**Comment by BeckerFrank** [comment]
> Let me rephrase that. Why are these "projects" around if it requires special knowledge to import the repository in order to have a clean workspace? What's the purpose of the projects? George
> […](#)

- mylyn.builds/org.eclipse.mylyn.hudson.tests/testdata
  the testdata is used for the setup of the Jenkins Test Instance (jobs/test-ant/config.xml)
- mylyn.reviews/tbr
  this is no longer active but I do not want to delete it yet so I first only do not do a Project Import for the projects
- mylyn.docs
  this is the previous location of https://github.com/eclipse-mylyn/org.eclipse.mylyn.docs/ and we can remove this now 
- /org.eclipse.mylyn.java.tests/target
  I found a few other cases where a Maven build left .project files in the build (target) folder and a Project Import added those projects
